avni-server-api/src/main/java/org/avni/web/api/GeneralEncounterApiController.java

```diff
@@ -85,9 +87,16 @@ public ResponseEntity<EncounterResponse> get(@PathVariable("id") String uuid) {
     @Transactional
     @ResponseBody
     public ResponseEntity post(@RequestBody ApiEncounterRequest request) {
-        Encounter encounter = new Encounter();
-        encounter.assignUUID();
+        Encounter encounter = createEncounter(request.getExternalId());
         try {
+            Individual individual = individualRepository.findByLegacyIdOrUuid(request.getSubjectId());
```

Comment:
This should find based on both values....it is using only one field.

Reply:
The lines after this does find by both UUID and LegacyId, have refactored code to introduce a local method that encapsulates that whole logic. Done that for all such instances.